providers/flagd/src/main/java/dev/openfeature/contrib/providers/flagd/resolver/process/storage/StorageStateDTO.java

```diff
@@ -0,0 +1,22 @@
+package dev.openfeature.contrib.providers.flagd.resolver.process.storage;
+
+
+import dev.openfeature.contrib.providers.flagd.resolver.process.model.FeatureFlag;
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.ToString;
+
+import java.util.Map;
+
+@Getter
+@ToString
+@EqualsAndHashCode
+public class StorageStateDTO {
+    private final StorageState storageState;
+    private final Map<String, FeatureFlag> changedFlags;
+
+    public StorageStateDTO(StorageState storageState, Map<String, FeatureFlag>  changedFlags) {
```

Comment:
a second constructor with just StorageState would be really beneficial, we should try to avoid passing null to a constructor

Reply:
This is correct will implement this.